### PR TITLE
[CondFormats/SiPixelObjects] Solve warning found by gcc 5.3.0

### PR DESCRIPTION
**Pull request by ghost**
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_8_1_X_2016-03-02-1100/CondFormats/SiPixelObjects
The behavior is unchanged, the parentheses makes clear the order of the logical operation while suppressing the warning.

### CODE REVIEW

**Comment by degano** [comment]
please test

**Comment by mmusich** [comment]
please test

**Comment by davidlt** [comment]
Wrong here. There is not (or I don't see) a point in converting `rawId()` to `1` or `0` (if operator is not overloaded). What author wanted was most likely `!=` instead of `==`. I.e. `(! roc->rawId() == detector.rawId)` to `(roc->rawId() != detector.rawId)`.

File: CondFormats/SiPixelObjects/src/SiPixelFrameConverter.cc [hunk 1]
```diff
@@ -53,7 +53,7 @@ int SiPixelFrameConverter::toCabling(
   for  (IT it = path.begin(); it != path.end(); ++it) {
     const PixelROC * roc = theMap->findItem(*it); 
     if (!roc) return 2;
-    if (! roc->rawId() == detector.rawId) return 3;
+    if ((! roc->rawId()) == detector.rawId) return 3;
```
Comment (VinInn) [inline CondFormats/SiPixelObjects/src/SiPixelFrameConverter.cc]: agree with @davidlt 
move ( after the !  and close it after the last )

**Comment by degano** [comment]
@VinInn @davidlt This change the current (but wrong?) behavior according to your suggestions.

**Comment by VinInn** [comment]
then do not modify and leave the warning there

**Comment by davidlt** [comment]
@degano correct, but these warnings are introduced in GCC for a reason and hiding them under the carpet is not a right way to deal with them.

**Comment by smuzaffar** [comment]
agree that https://github.com/degano/cmssw/commit/a73b83b9eaa0c3f398eace753f73ba6f82728d8f is the correct change (though it changes current behaviour). Lets test it.

**Comment by smuzaffar** [comment]
please test

File: CondFormats/SiPixelObjects/src/SiPixelFrameConverter.cc (hunk 1 above)
Comment (dkotlins) [inline CondFormats/SiPixelObjects/src/SiPixelFrameConverter.cc]: Vincenzo,
If you give me a day I can fix it.
I also do not know what the intention was but I think I can figure it out.
I could have it ready by Monday.
Or shall I wait that this pull request is done?
Danek

**Comment by mmusich** [comment]
@ggovi ping

**Comment by davidlange6** [comment]
Adding @bartokm, @dkotlins to possibly help answer the questions in this pull request.

**Comment by bartokm** [comment]
Unfortunately I was not involved in the development of this code, besides that I'm not sure what is the question...

**Comment by dkotlins** [comment]
Looks fine to me. Please proceed.
Danek

**Comment by mmusich** [comment]
+1

**Comment by ggovi** [comment]
+1

**Pull request merged**